Assets/MixedRealityToolkit/Utilities/BuildAndDeploy/UwpPlayerBuildTools.cs

```diff
@@ -86,7 +86,14 @@ async void PostBuildAction(IBuildInfo innerBuildInfo, BuildReport buildReport)
                     var uwpBuildInfo = innerBuildInfo as UwpBuildInfo;
                     Debug.Assert(uwpBuildInfo != null);
                     UwpAppxBuildTools.AddCapabilities(uwpBuildInfo);
-                    UwpAppxBuildTools.UpdateAssemblyCSharpProject(uwpBuildInfo);
+
+#if !UNITY_2019_1_OR_NEWER
+                    if (EditorUserBuildSettings.wsaGenerateReferenceProjects &&
```

Comment:
Nit: It feels weird for us to double guard this (i.e. once prior to invocation time, and again within the function itself. It feels like the function should either handle it for all callers (i.e. it abstracts out that complication) or it should define formally that it's responsibilities don't include checking, so callers should check.

Reply:
IMHO i would also just wrap the method `UpdateAssemblyCSharpProject` in `UwpAppxBuildTools.cs` and the call here in a `#if !UNITY_2019_1_OR_NEWER` preprocessor guard. With that and the preprocessor guard in the layout, the in Untiy 2019 useless code path would just be gone.

Reply:
Makes sense! I'll update this